scipy/linalg/tests/test_sketches.py

```diff
@@ -45,6 +45,7 @@ class TestClarksonWoodruffTransform:
 
     # Test vector with norm ~1
     x = rng.random((n_rows, 1)) / np.sqrt(n_rows)
+    del rng  # Not deterministic in pytest-run-parallel
```

Comment:
I think it's because this test is written in a weird way, normally this code would go into `setup_method`. Fine with me to leave as is though.

Reply:
Yeah, the way it's written it's just working around not being able to have a `__init__` method. My addition just makes it clear that the rng should not be reused in the test methods.

Note that `setup_method` to my understanding would be problematic with `pytest-run-parallel`.

Reply:
> Note that `setup_method` to my understanding would be problematic with `pytest-run-parallel`.

It shouldn't be AFAIK, it's used extensively in the SciPy test suite. It should run just once, and since all it does is set up arrays/objects as attributes for read-only usage, it should be fine.

Reply:
> It should run just once, and since all it does is set up arrays/objects as attributes for read-only usage, it should be fine.

I've been seeing it used a lot to set up a random state (like in many points that are cleaned up by this PR), a temporary directory, a temporary file handle, etc.